### PR TITLE
Fix linked clp result creation

### PR DESCRIPTION
**Pull request by joernweissenborn**
This fixes the issue that CLP in linked data not the same because they were calculated individually in the result creation.

### CODE REVIEW

File: glotaran/optimization/objective.py [hunk 1]
```diff
@@ -30,36 +30,54 @@ def __init__(self, model: ExperimentModel):
...
+        reduced_matrices: list[OptimizationMatrix],
+        estimations: list[OptimizationEstimation],
+    ) -> list[OptimizationEstimation]:
+        return [
+            e.resolve_clp(m.clp_axis, r.clp_axis, i, self._model.clp_relations)
+            for e, m, r, i in zip(estimations, matrices, reduced_matrices, self._data.global_axis)
+        ]
+
+    def calculate_global_penalty(self) -> ArrayLike:
+        _, _, matrix = OptimizationMatrix.from_global_data(self._data)  # type:ignore[arg-type]
```
[the remark this answers is not in the document]
Comment (s-weigand) [reply glotaran/optimization/objective.py:64]: Seams like sourcery does not understand how `_` is used in python.

File: glotaran/optimization/objective.py [hunk 2]
```diff
@@ -70,46 +88,100 @@
...
+    def get_result(self) -> dict[str, xr.Dataset]:
+        return (
+            self.create_unlinked_result()
+            if isinstance(self._data, OptimizationData)
+            else self.create_linked_result()
+        )
+
+    def create_result_dataset(self, label: str, data: OptimizationData) -> xr.Dataset:
+        assert isinstance(data.model.data, xr.Dataset)
+        dataset = data.model.data.copy()
```
[the remark this answers is not in the document]
Comment (s-weigand) [reply glotaran/optimization/objective.py:100]: As per [xarray docs](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.copy.html) Sourcery got this right.
Either we should use `data.model.data.copy(deep=True)` or omit the `.copy()`.

**Comment by jsnel** [comment]
Not to future self (@jsnel ) - as suggested by the bots in this PR, review in greater detail all .copy operations (at a later time).

**Review by jsnel — APPROVED** [review]
Unit tests are passing.
https://github.com/glotaran/pyglotaran/actions/runs/7595901170

Coverage improved
https://app.codecov.io/github/glotaran/pyglotaran/pull/1418

The link-clp-sim-staging.ipynb notebook was run to verify thinks are working correctly in terms of clp linking (at least within the scope of that notebook).

Code will be further evaluated for other test cases on staging, should be ok to merge in, so further evaluation can start.

**Pull request merged**